### DISCUSSION TITLE
DuckDB too slow to store sensor data at a high rate?

### DISCUSSION THREAD

**Discussion by samsnori** [duckdb/duckdb#14743] [Q&A]
I'm looking into DuckDB to store position data persistently (on disk) from sensors at a "high rate". I'm receiving messages with: position (float4 x,y,z), entity_id (uint64_t), sensor_id (uint64_t), timestamp, at a rate of ~100Hz and trying to figure out how to store these in a DuckDB database w/o clogging my pipeline.  I might receive data from several sensors (~20).

I did a quick test with the C-API and noticed that a prepared statement takes ~200ms when using disk storage and ~0.7ms when using in memory storage. Both are too slow and even the in memory solution would mean I need to buffer + drop messages to keep up. I'm curious if someone might know a solution to store this information?

**Comment by prashanthellina** [comment]
Using the appender API, I was able to get to 600k rows inserted per second.

**Reply by chu-k** [reply]
Hi, out of curiosity, are you flushing the appender on each row insert?

**Reply by prashanthellina** [reply]
I don't flush at all. I let the appender choose when to flush to disk.

**Comment by samsnori** [comment]
Awesome, I've just learned about the appender API, thanks!
